### PR TITLE
Add check-dependent-jobs action

### PR DESCRIPTION
**Pull request by reakaleek**
will be handy for testing in this repo where the `test` status check is expected

### CODE REVIEW

File: .github/workflows/test-check-dependent-jobs.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./check-dependent-jobs
+        id: check
+        with:
+          needs: '{ "job-a": { "result": "success" }, "job-b": { "result": "success" } }'
+      - name: assert
```
Comment (v1v) [inline .github/workflows/test-check-dependent-jobs.yml:29]: **nit**: maybe the expected value in the name of the step?

File: .github/workflows/test-check-dependent-jobs.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        run: test "${{ steps.check.outputs.is-success }}" = "true"
+      - uses: ./check-dependent-jobs
+        id: check-2
+        with:
+          needs: '{ "job-a": { "result": "success" }, "job-b": { "result": "failure" } }'
+      - name: assert
```
Comment (v1v) [inline .github/workflows/test-check-dependent-jobs.yml:35]: ditto

File: check-dependent-jobs/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+## About
```
Comment (v1v) [inline check-dependent-jobs/README.md:1]: I think the title is not following the pattern `github-commposite-name

```suggestion
# check-dependent-jobs
```

File: check-dependent-jobs/action.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+name: check-dependent-jobs
+description: |
+  Evaluates the combined the status results of the provided needs context.
+inputs:
+  needs:
```
Comment (v1v) [inline check-dependent-jobs/action.yml:5]: wonder if `jobs` input is more meaningful

File: check-dependent-jobs/action.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  is-success:
+    description: The evaluated result of all provided jobs in the needs context.
+    value: ${{ steps.test.outputs.is-success }}
+  status:
+    description: One of success or failure.
+    value: ${{ steps.test.outputs.status }}
+runs:
+  using: composite
+  steps:
+    - id: test
```
Comment (v1v) [inline check-dependent-jobs/action.yml:18]: Do you think we should support this step as OS agnostic?

Comment (v1v) [reply check-dependent-jobs/action.yml:18]: I've just found some issues when the needs contains multiple outputs, and therefore the bash fails

File: .github/workflows/test-check-dependent-jobs.yml (hunk 2 above)
Comment (reakaleek) [reply .github/workflows/test-check-dependent-jobs.yml]: https://github.com/elastic/oblt-actions/pull/16/commits/059a02bc851bbb029a31c59ab31184948683aabe

File: .github/workflows/test-check-dependent-jobs.yml (hunk 1 above)
Comment (reakaleek) [reply .github/workflows/test-check-dependent-jobs.yml]: https://github.com/elastic/oblt-actions/pull/16/commits/059a02bc851bbb029a31c59ab31184948683aabe

File: check-dependent-jobs/action.yml (hunk 5 above)
Comment (reakaleek) [reply check-dependent-jobs/action.yml:18]: Can you please elaborate on this?

File: check-dependent-jobs/action.yml (hunk 4 above)
Comment (reakaleek) [reply check-dependent-jobs/action.yml]: https://github.com/elastic/oblt-actions/pull/16/commits/e14f866c4f08a0d77f9463377053cb7c8d79c2a3

File: .github/workflows/test-check-dependent-jobs.yml (hunk 2 above)
Comment (reakaleek) [reply .github/workflows/test-check-dependent-jobs.yml]: https://github.com/elastic/oblt-actions/pull/16/commits/6d402476a15d07aff522ba1c369a3fabb7312894

File: check-dependent-jobs/action.yml (hunk 5 above)
Comment (reakaleek) [reply check-dependent-jobs/action.yml:18]: https://github.com/elastic/oblt-actions/pull/16/commits/0e8037f0f1e1166e82fc37a50c7302dd104fe3ca

**Review by v1v — APPROVED** [review]

**Comment by reakaleek** [comment]
@v1v I saw https://github.com/elastic/app-obs-dev/pull/55.

Do we need to change this?

**Comment by v1v** [comment]
> I saw https://github.com/elastic/app-obs-dev/pull/55.

It's a private repo 😅 

I'm not sure, I guess we could potentially support some extra arguments:

```yaml
with:
  compare: all
  status: failure

with:
  compare: any
  status: failure

with:
  compare: none
  status: success
```

or something along those lines...

**Comment by reakaleek** [comment]
@v1v woops. Deleted the comment

**Comment by reakaleek** [comment]
O

> I'm not sure, I guess we could potentially support some extra arguments:
> 
> ```yaml
> with:
>   compare: all
>   status: failure
> 
> with:
>   compare: any
>   status: failure
> 
> with:
>   compare: none
>   status: success
> ```
> 
> or something along those lines...

Sounds like it could be an enhancement in a follow-up.

I will merge this for now as we have some usages of this with this functionality already.

**Pull request merged**